launcher/minecraft/mod/tasks/LocalResourceParse.cpp

```diff
@@ -63,17 +57,22 @@ PackedResourceType identify(QFileInfo file){
         } else if (ShaderPackUtils::validate(file)) {
             qDebug() << file.fileName() << "is a shader pack";
             return PackedResourceType::ShaderPack;
+        } else if (ModUtils::validate(file)) {
+            // mods should be already filtered out by the API response
+            qDebug() << file.fileName() << "is a mod";
+            return PackedResourceType::Mod;
```

Comment:
This identify function is used in places where there is no API to identify anything. like importing an arbitrary raw archive.

Mods can still cintain resource and data packs and must be the first to identify int eh logic of this function. there is no filtering.

Reply:
reverted back the file(I did not see any uses for that except for flame instance creation; I may not look hard enough).